### PR TITLE
(paperless-ngx) fix(charts/paperless-ngx): Fix Redis External Secret

### LINKED ISSUE #8: [paperless-ngx] existingSecret doesn't work for redis
### Name and Version

fmjstudios/paperless-ngx 0.2.0

### What architecture are you using?

arm64

### What steps will reproduce the bug?

Deploy helm chart with existing secret for paperless 

### Are you using any custom parameters or values?

```
paperless:
  domain: <DOMAIN>
  redis:
    host: redis
    existingSecret: redis-secret
tika:
  image:
    tag: 2.9.2.1
redis:
  enabled: false
postgresql:
  enabled: false
```

### What is the expected behavior?

Secret is picked up and fed into ENV of the container

### What do you see instead?

Pod paperless-ngx-0 cannot be started because

secret "paperlessngx-paperless-ngx-redis-auth" not found

### Additional information

the required secrets file is not built, although in '_secrets.tpl' you can already find a special treatment for existing secrets...

### PR DESCRIPTION
**Pull request by MyChaOS87**
env PAPERLESS_REDIS was always referencing a specific secret which was never built when usind an externalSecret for Redis, rendering paperless.redis.existingSecret useless, although there was some code to create an URI out of the existing secret.
Now the URI is constructed by using dependent ENV variables.

#### Which issue this PR fixes

- fixes #8 

#### Checklist

- [x] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `(paperless-ngx)`)

### CODE REVIEW

**Comment by FMJdev** [comment]
Hey @MyChaOS87,

I have just deployed a fix in version `0.2.1` mitigating you issue. Check out [the commit](https://github.com/fmjstudios/helm/commit/39cc552fffdf17bdc7c28c1cbce5289cd604884c) for more in-depth information. 

### TL;DR

The `existingSecret` now requires an existing Secret with a valid Redis connection URI since only using keys from a specific Secret would always require me to create a new one with a full URI later, since setting the environment variables directly isn't good practice. This inadvertently creates duplicated or overlapping secrets - a pain to deal with at scale.

I have chosen not to merge your PR since these new changes make them obsolete and I wan't keep the number of environment variables at a minimum since there are enough in Kubernetes. Thank you for your work nonetheless! 👍🏼

**Comment by MyChaOS87** [comment]
Although I see your reasoning, this now renders the chart basically unusable for me. As I now have to render the URI into a sealed secret, where at secret creation time I should not need to know the hostname of my redis instance... I know that this comes from the use of that URI directly in paperless-ngx (which in my opinion is a big flaw on that side), but encoding a full URI in a (sealed-) secret also seems way off to me.

**Comment by FMJdev** [comment]
Responded in #10. Please restrict further conversation to the (reopened) issue.
